scripts/update-cdn-urls.js

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env node
+/**
+ * @fileoverview
+ * Updates the CDN URLs in the README.md to the latest major version. Does
+ * nothing if the README.md is already up-to-date.
+ *
+ * It assumes the major version in the README is either equal to the current
+ * package major version, or the package minor version minus 1. If not the
+ * behaviour of this script should be consider undefined.
+ */
+
+const fs = require("fs");
+const path = require("path");
+const getMajorVersion = require("semver/functions/major");
```

Comment:
Why use a third party library and not just to use `parseInt(manifest.version.split('.')[0])`? The replacement uses `majorVersion - 1` anyways.

Reply:
> Why are you using a third party library and not just to use `parseInt(manifest.version.split('.')[0])`?

Because why not, it should be more robust - though I agree there is not much that can go wrong here - and it's definitely easier to parse `getMajorVersion` than `parseInt(manifest.version.split('.')[0])`. If you have a particular concern about using a dependency (e.g. security, install speed, something else) I'd happily change it. For what it's worth we use [the same dependency in our release action](https://github.com/simple-icons/release-action/blob/4c952419b4374cce8d114104ac07679adf655a6c/package.json#L28).

If we do decide to change it, my vote would go towards having a local function named `getMajorVersion` that, in some form, does `parseInt(manifest.version.split('.')[0])`.


> The replacement uses `majorVersion - 1` anyways.

Not sure how this relates 🤔 Can you elaborate?

Reply:
> Not sure how this relates thinking Can you elaborate?

That the update is just to increase an integer. I think it is not necessary to add a library for this if we are not going to generate more complex version names.

> Because why not, it should be more robust

How could be this more robust? I think that use a third party library for this simple task is just obfuscate the code needlessly.

> If we do decide to change it, my vote would go towards having a local function named getMajorVersion that, in some form, does `parseInt(manifest.version.split('.')[0])`.

I like that idea :+1:

Reply:
I will update this on the basis that indeed this is rather simple and using a dependency does have (other) drawbacks.


---


> That the update is just to increase an integer. I think it is not necessary to add a library for this if we are not going to generate more complex version names.

I don't agree that what you do with the date should determine how you get it. For example, we don't need to parse the JSON (from Node's standard library) in order to get the version number, we can also use the RegExp `/"version":\s*"([^"]*)"/` or go to the exact character-index we expect to get it. But parsing the JSON is easier to understand and work with.

> How could be this more robust? I think that use a third party library for this simple task is just obfuscate the code needlessly.

I completely agree that in this case there does not seem to be much to be worried about w.r.t. robustness. However, I don't see how using a third-party library obfuscates anything. The function name is descriptive, simple, and easy to understand provided you're familiar with SemVer (if you're not, library vs. local code is not going to make a difference). Moreover, this is JavaScript so you can just read the source code of the library (as opposed to e.g. a `.jar` file for Java).